pkg/lakom/resolvetag/admission.go

```diff
@@ -125,81 +133,195 @@ func (h *handler) GetLogger() logr.Logger {
+		if err != nil {
+			return nil, err
 		}
+		extension.Spec.Deployment.AdmissionDeployment.RuntimeCluster.Helm.OCIRepository.Ref = &resolved
 	}
 
-	for idx, c := range p.Spec.Containers {
-		image, err := h.handleContainer(ctx, c.Image, kcr, logger.WithValues("container", c.Name))
+	if extension.Spec.Deployment != nil &&
+		extension.Spec.Deployment.AdmissionDeployment != nil &&
+		extension.Spec.Deployment.AdmissionDeployment.VirtualCluster != nil &&
+		extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm != nil &&
+		extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm.OCIRepository != nil {
+		resolved, err := h.resolveArtifact(ctx, extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm.OCIRepository.GetURL(), kcr, logger)
 		if err != nil {
-			return err
+			return nil, err
 		}
-		if image != "" {
-			p.Spec.Containers[idx].Image = image
+		extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm.OCIRepository.Ref = &resolved
+	}
+
+	if extension.Spec.Deployment != nil &&
+		extension.Spec.Deployment.ExtensionDeployment != nil &&
+		extension.Spec.Deployment.ExtensionDeployment.Helm != nil &&
+		extension.Spec.Deployment.ExtensionDeployment.Helm.OCIRepository != nil {
+		resolved, err := h.resolveArtifact(ctx, extension.Spec.Deployment.ExtensionDeployment.Helm.OCIRepository.GetURL(), kcr, logger)
+		if err != nil {
+			return nil, err
 		}
+		extension.Spec.Deployment.ExtensionDeployment.Helm.OCIRepository.Ref = &resolved
 	}
 
-	for idx, ec := range p.Spec.EphemeralContainers {
-		image, err := h.handleContainer(ctx, ec.Image, kcr, logger.WithValues("ephemeralContainer", ec.Name))
+	return json.Marshal(extension)
+}
+
+func (h *handler) handlePod(ctx context.Context, pod corev1.Pod) ([]byte, error) {
+	var (
+		logger = h.logger.WithValues("pod", client.ObjectKey{Name: pod.Name})
+		kcr    = utils.NewLazyKeyChainReaderFromPod(ctx, h.reader, &pod, h.useOnlyImagePullSecrets)
+	)
+
+	for idx, ic := range pod.Spec.InitContainers {
+		resolved, err := h.resolveArtifact(ctx, ic.Image, kcr, logger)
 		if err != nil {
-			return err
+			return nil, err
 		}
-		if image != "" {
-			p.Spec.EphemeralContainers[idx].Image = image
+
+		pod.Spec.InitContainers[idx].Image = resolved
+	}
+	for idx, c := range pod.Spec.Containers {
+		resolved, err := h.resolveArtifact(ctx, c.Image, kcr, logger)
+		if err != nil {
+			return nil, err
 		}
+
+		pod.Spec.Containers[idx].Image = resolved
+	}
+	for idx, ec := range pod.Spec.EphemeralContainers {
+		resolved, err := h.resolveArtifact(ctx, ec.Image, kcr, logger)
+		if err != nil {
+			return nil, err
+		}
+
+		pod.Spec.EphemeralContainers[idx].Image = resolved
+	}
+
+	return json.Marshal(pod)
+}
+
+// getURL returns the fully-qualified OCIRepository URL of the image.
+func getURL(img *seedmanagementv1alpha1.Image) string {
+	ref := *img.Repository
+
+	if img.Tag != nil {
+		ref = ref + ":" + *img.Tag
```

Comment:
I guess the tag can contain sha256 digest, if that is the case - the ref should be 
```go
ref + "@" + *img.Tag
```